src/contracts/core/DelegationManager.sol

```diff
@@ -483,6 +501,17 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         return approverDigestHash;
     }
 
+    /** 
+     * @notice Returns 'true' if the `staker` can immediately undelegate without queuing a new withdrawal OR if the staker is already undelegated,
+     * and 'false' otherwise
+     * @dev A staker can only undelegate if they have no "active" shares in EigenLayer and are not themselves an operator
+     */
+    function canUndelegate(address staker) public view returns (bool) {
+        return (!isOperator(staker) &&
```

Comment:
This should also have a condition for `isDelegated(staker)` otherwise, this will return true when they're already undelegated

Reply:
yeah, good point. it has a comment which notes the current behavior, but I agree your suggested change aligns with the stricter behavior added to the `undelegate` function (reverting rather than being a no-op if the staker is already undelegated)

Reply:
or this function could be renamed to something like `hasActivelyDelegatedShares` ?
I think this would make much more sense with the current behavior, actually.
any thoughts on this @stevennevins ?

Reply:
> or this function could be renamed to something like `hasActivelyDelegatedShares` ?
> I think this would make much more sense with the current behavior, actually.
> any thoughts on this @stevennevins ?

Yeah, I like that better too

Reply:
This ended up being a touch more complicated than I was thinking -- can you do a bit of a spot-check on the commit? https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/be4a017e0c0b3535318a25f759601bcf07bc22a0

Reply:
I added one more subsequent commit that just deletes the function 😄 -- see https://github.com/Layr-Labs/eigenlayer-contracts/pull/176/commits/96cbe826c83cf1c7ec531a4f8b01b4ecfa7e7ab1
I think the logic is both clearer and cleaner now